osidb/models.py

```diff
@@ -1855,30 +1852,43 @@ def _validate_sofware_collection(self):
                 "does not match any valid collection",
             )
 
-    def _validate_exceptional_affectedness_resolution(self):
+    def _validate_historical_affectedness_resolution(self):
         """
-        Alerts that an old flaw have empty affectedness.
-        (Only accepts WONTFIX and DEFER as resolution, validated by other validation)
+        Alerts that an old flaw has an affectedness/resolution combination that is now invalid,
+        but was valid in the past.
         """
-        valid_resolutions = [
-            Affect.AffectResolution.WONTFIX,
-            Affect.AffectResolution.DEFER,
-        ]
         if (
-            self.affectedness == Affect.AffectAffectedness.NOVALUE
-            and self.resolution in valid_resolutions
+            self.resolution
+            in AFFECTEDNESS_HISTORICAL_VALID_RESOLUTIONS[self.affectedness]
         ):
-            self.alert(
-                "flaw_exceptional_affect_status",
-                f"Affect ({self.uuid}) for {self.ps_module}/{self.ps_component} is in "
-                "a exceptional state having no affectedness.",
-            )
+            # Don't allow new records with historical combinations, but when modifying existing
+            # data, only throw an alert
+            if self._state.adding:
+                raise ValidationError(
+                    f"{self.resolution} is not a valid resolution for {self.affectedness}."
+                )
+
+            old_affect = Affect.objects.get(uuid=self.uuid)
+            if (
+                self.affectedness != old_affect.affectedness
+                or self.resolution != old_affect.resolution
+            ):
```

Comment:
This is one step closer but I probably did not explain myself well enough :smile:

Great that we now cannot create a new affect with the invalid values but still it is possible to assign invalid values to an existing one. We should prevent all new invalid combinations, either on creation or on update.

Only when the affectedness and resolution combination was already invalid and we do not touch it but eg. change the affect impact, we should create an alert but permit the change.